pkg/common/hashmap/strhashmap.go

```diff
@@ -170,15 +171,36 @@ func (m *StrHashMap) encodeHashKeys(vecs []*vector.Vector, start, count int) {
 	}
 }
 
-func fillStringGroupStr(m *StrHashMap, vec *vector.Vector, n int, start int) {
+func HashKeys(keys []byte, bytes []byte) []byte {
```

Comment:
no, you are not going to use make here.  

you are just serialize keys to a bytes, correct?   why not just length prefix + data?   You should compute the length you need, call one append at most.

Reply:
the hash function need the len(keys) at least 16bytes, but I can't modify the source keys, however I will give you another way